src/frontend/hooks/useKeepPlaystreaksInSync.ts

```diff
@@ -0,0 +1,36 @@
+import { questPlayStreakSyncState } from '@hyperplay/quests-ui'
+import { useEffect, useRef } from 'react'
+import { useQueryClient } from '@tanstack/react-query'
+
+export function useKeepPlaystreaksInSync(appName: string) {
+  const syncInitializedRef = useRef(false)
+  const queryClient = useQueryClient()
+
+  useEffect(() => {
+    return () => {
+      if (syncInitializedRef.current) {
+        questPlayStreakSyncState.clearAllTimers()
+      }
+    }
+  }, [])
+
+  useEffect(() => {
+    const initSync = async () => {
+      try {
+        questPlayStreakSyncState.init({
```

Comment:
This is the only place it is used, so it is simpler to just call init right before. I initially explored adding it to `StoreController`, but the `StoreController` `useEffect` executes after this useEffect, which breaks `keepProjectQuestsInSync`

Reply:
how interesting, in my implementation of this in the store I used the StoreController approach 

https://github.com/HyperPlay-Gaming/hyperplay-store/pull/228/files#diff-c4fdb34f33780698be8e19adc9cda4a65464bcd97bdc4a7ed870e888a3c5e42d

tbh I think this way is simpler and does the work too

Reply:
maybe we should move this into `@hyperplay/quests-ui` if it will be used on the store as well